docs/modules/ROOT/pages/gsn-bouncers.adoc

```diff
@@ -209,31 +206,44 @@ function _postRelayedCall(bytes memory context, bool, uint256 actualCharge, byte
 }
 ----
 
-This is required to protect the contract from exploits (this is really similar to how ether is locked in Ethereum transactions).
+The maximum amount of tokens required is transferred to protect the contract from exploits (this is really similar to how ether is locked in Ethereum transactions).
+
+NOTE: The gas cost estimation is not 100% accurate, we may tweak it further down the road.
 
-Please note how the gas cost estimation is not 100% accurate, we may tweak it further down the road.
+NOTE: Internal `_preRelayedCall` and `_postRelayedCall` functions are used instead of public `preRelayedCall` and `postRelayedCall` functions. This prevents them from being called by non-RelayHub contracts. 
+Always use `_preRelayedCall` and `_postRelayedCall` functions.
 
-NOTE: `_preRelayedCall` and `_postRelayedCall` are used instead of preRelayedCall and postRelayedCall. This prevents them from being called by non-relayhub. Always use _pre and _post methods.
+=== How to use GSNBouncerERC20Fee
 
-=== How to use it
+Your GSN recipient contract needs to inherit from `GSNRecipient` and `GSNBouncerERC20Fee` along with appropriate xref:access-control.adoc[access control] (for token minting), set the token details in the constructor of `GSNBouncerERC20Fee` and create a public `mint` function suitably protected by your chosen access control as per the following sample code (which uses the https://docs.openzeppelin.com/contracts/2.x/api/access#MinterRole[MinterRole]):
 
-Create your contract using the following:
+NOTE: The token must have decimals of 18 to match that of ether, due to the baked-in exchange rate of 1:1.
 
 [source,solidity]
 ----
-  mycontract is GSNRecipient, GSNBouncerERC20Fee {
-    constructor(name symbol decimals)
+contract MyContract is GSNRecipient, GSNBouncerERC20Fee, MinterRole {
+    constructor() public GSNBouncerERC20Fee("FeeToken", "FEE", 18) {
+    }
 
-    mint() {
-      _mint()
+    function mint(address account, uint256 amount) public onlyMinter {
+        _mint(account, amount);
     }
-  }
+}
 ----
 
-== Create your custom Bouncer [optional, for power users]
+== Custom Bouncer
+
+You can create your own Custom Bouncer!  For example, your Custom Bouncer could use a specified token to pay for transactions with a custom exchange rate to ether.  Alternatively you could issue users who subscribe to your dapp ERC721 tokens and accounts holding the subscription token could use your contract for free as part of the subscription.  There are lots of potential options for your Custom Bouncer.
 
-You can use 'GSNBouncerBase' as an example to guide your Bouncer implementation.
+Your Custom Bouncer can inherit from `GSNBouncerBase` and must implement the `acceptRelayedCall` function.
 
-The only thing you must do is extend from `GSNRecipient` and implement the accept method.
+Depending on the logic for your Custom Bouncer, you may need to implement `_postRelayedCall` and `_preRelayedCall`.
```

Comment:
I think we should explain how the charge estimates work, and why they work that way: that `pre` will tell you the maximum possible charge, and later `post` will tell you the real amount. Or point towards `GSNBouncerERC20Fee` as an example of how to deal with it.

Reply:
I have pointed to the GitHub repo 2.4.0 versions of `GSNBouncerERC20Fee` and `GSNBouncerSignature`.  Is there a better way to do this?